per.py has > 4e-5 error
